### ISSUE TITLE
JSONC support

### ISSUE THREAD

**Issue by woojiq** [helix-editor/helix#5452]
__EDIT: collapsible sections for better readability__

<details><summary>Problem</summary>

## Problem

Json highlighting works out of the box, but then I opened my Jsonc config file and the text was greyed out. I did some research and realized that this is the correct default behavior because there is no `file-types = [..., "jsonc"]` for json language: https://github.com/helix-editor/helix/blob/b7ca0581d81f0ac80418a3e052df2bd5d9f71ad4/languages.toml#L141

Then I added this line:
``` toml
[[language]]
name = "json"
file-types = ["jsonc", "json"]
```

After this _jsonc_ was also colored but comments was bold red (means error):
![image](https://user-images.githubusercontent.com/97038258/211206738-21080972-d09d-4514-ad41-ca6cfd733c6a.png)

I looked in _vscode json server_ and found that I can pass `languageId = "jsonc"` to disable comment warnings:  
[Link to lines](https://github.com/microsoft/vscode/blob/4f69cdf95a12cef48d405b38bf7812a7f297c310/extensions/json-language-features/server/src/jsonServer.ts#L324-L326)
``` ts
const documentSettings: DocumentLanguageSettings = textDocument.languageId === 'jsonc' ? { comments: 'ignore', trailingCommas: 'warning' } : { comments: 'error', trailingCommas: 'error' };
```
</details>

<details><summary>What I tried</summary>

### Attempt 1
Pass `language-id = "jsonc"` as an argument to the _json_ language server so that it works for both formats:
``` toml
[[language]]
name = "json"
file-types = ["jsonc", "json"]
language-server = { command = "vscode-json-language-server", args = ["--stdio"], language-id = "jsonc" }
```
__Result__:  nothing changes. The colors are the same as in the picture.

### Attempt 2
Create new language _jsonc_ and also pass _lang-id_ argument to language-server:
``` toml
[[language]]
name = "jsonc"
scope = "source.jsonc"
file-types = ["jsonc"]
roots = []
language-server = { command = "vscode-json-language-server", args = ["--stdio"], language-id = "jsonc" }
config = { "provideFormatter" = true, "json" = { "keepLines" = { "enable" = true } } }
indent = { tab-width = 2, unit = "  " }
```
__Result__:  gray text without any color. Adding next doesn't help:
``` toml
[[grammar]]
name = "jsonc"
source = { git = "https://github.com/tree-sitter/tree-sitter-json", rev = "65bceef69c3b0f24c0b19ce67d79f57c96e90fcb" }
```
</details>

## Question
How to fix one of my attempts to make it work or tell another approach to disable comments (and trailingCommas) warnings in jsonc?

__P.S.__ This is my first experience with all this stuff that vscode and the IDEs do themselves, so I might be missing something.

**Comment by gabydd** [comment]
We need a different tree-sitter parser for jsonc, the one tree-sitter neovim uses is: https://gitlab.com/WhyNotHugo/tree-sitter-jsonc.git

**Comment by woojiq** [comment]
Should I also write own queries? I thought tree-sitter does it for me. Btw, I couldn't setup this properly and dont have enough patients to continue. Maybe 'Close as not planned'?  
At first I thought it was just about passing the `language-id` argument to vscode-ls, but then I found out about all this tree-sitter/query stuff. And now I'm happy with helix default setup;)

**Comment by the-mikedavis** [comment]
Yeah this would also need custom queries. I would expect that the queries would be very similar to the ones in `runtime/queries/json` though since the syntax for jsonc is just a little different.

I'll leave this open but tweak the title a little - someone might come along and contribute the configuration and queries for JSONC support

**Comment by archseer** [comment]
Closed by #5509
